### PR TITLE
Update ExceptionDispatchInfo documentation

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
/cc @janvorli @dotnet/docs @carlossanlop @mangod9

### CODE REVIEW

File: xml/System.Runtime.ExceptionServices/ExceptionDispatchInfo.xml [hunk 1]
```diff
@@ -126,10 +126,9 @@
         <Parameter Name="source" Type="System.Exception" Index="0" FrameworkAlternate="net-5.0" />
       </Parameters>
       <Docs>
-        <param name="source">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="source">The unthrown <see cref="T:System.Exception"/> instance.</param>
```
Comment (gewarren) [inline xml/System.Runtime.ExceptionServices/ExceptionDispatchInfo.xml:129]:
```suggestion
        <param name="source">The unthrown exception.</param>
```

Parameter and return descriptions should avoid mentioning the data type. See https://github.com/dotnet/dotnet-api-docs/wiki/Parameters%3A-Methods-and-constructors. The data type is shown automatically above the description.

File: xml/System.Runtime.ExceptionServices/ExceptionDispatchInfo.xml [hunk 2]
```diff
@@ -126,10 +126,9 @@
...
+        <summary>Stores the current stack trace into the specified <see cref="T:System.Exception"/> instance.</summary>
+        <returns>The <paramref name="source" /> exception instance.</returns>
```
Comment (gewarren) [inline xml/System.Runtime.ExceptionServices/ExceptionDispatchInfo.xml:131]:
```suggestion
        <returns>The <paramref name="source" /> exception instance with the stack trace included.</returns>
```

File: xml/System.Runtime.ExceptionServices/ExceptionDispatchInfo.xml [hunk 3]
```diff
@@ -126,10 +126,9 @@
         <Parameter Name="source" Type="System.Exception" Index="0" FrameworkAlternate="net-5.0" />
       </Parameters>
       <Docs>
-        <param name="source">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
```
Comment (gewarren) [inline xml/System.Runtime.ExceptionServices/ExceptionDispatchInfo.xml:132]: I don't think we're supposed to delete these. @carlossanlop Does it matter?

Comment (AaronRobinsonMSFT) [reply xml/System.Runtime.ExceptionServices/ExceptionDispatchInfo.xml:132]: I added them back. I think the tool will keep adding them. I forgot to revert this change in the PR.

**Pull request merged**